lasagne/layers/noise.py

```diff
@@ -15,12 +17,58 @@
 
 
 class DropoutLayer(Layer):
-    def __init__(self, incoming, p=0.5, rescale=True, **kwargs):
-        super(DropoutLayer, self).__init__(incoming, **kwargs)
+    """Dropout layer [1]_,[2]_
+
+    Sets values to zero with probability p. See notes for disabling dropout
+    during testing.
+
+    Parameters
+    ----------
+    input_layer : `Layer` instance
+        The layer from which this layer will obtain its input
+    p : float or scalar tensor
+        The probability of setting a value to zero
+
+    Notes
+    ----------
+    The dropout layer is a regularizer that randomly sets input values to
+    zero, see [1,2] for a why this might improve generalization.
+    During training you should set deterministic to false and during
+    testing you should set deterministic to true.
+
+    If rescale is true the input is scaled with input / (1-p) when
+    deterministic is false, see [1,2] for further discussion. Note that this
+    implementation scales the input at training time.
+
+    References
+    ----------
+    [1] Hinton, G., Srivastava, N., Krizhevsky, A., Sutskever, I.,
+    Salakhutdinov, R. R. (2012).
+    Improving neural networks by preventing co-adaptation of feature detectors.
+    arXiv Preprint, 1207.0580(Hinton, Geoffrey E., et al.
+
+    [2] Srivastava Nitish, Hinton, G., Krizhevsky, A., Sutskever,
+    I., & Salakhutdinov, R. R. (2014).
+    Dropout: A Simple Way to Prevent Neural Networks from Overfitting.
+    Journal of Machine Learning Research, 5(Jun)(2), 1929-1958.
+    """
+    def __init__(self, input_layer, p=0.5, rescale=True):
+        super(DropoutLayer, self).__init__(input_layer)
         self.p = p
         self.rescale = rescale
 
-    def get_output_for(self, input, deterministic=False, **kwargs):
+    def get_output_for(self, input, deterministic=False, *args, **kwargs):
```

Comment:
The `*args` have been removed some time ago. Are you working from the latest `master`?